bundles/org.openhab.binding.tr064/src/main/java/org/openhab/binding/tr064/internal/phonebook/PhonebookProfileFactory.java

```diff
@@ -58,12 +81,37 @@
 
     @Override
     public Collection<ProfileTypeUID> getSupportedProfileTypeUIDs() {
-        return Collections.singleton(PhonebookProfile.PHONEBOOK_PROFILE_TYPE_UID);
+        return Set.of(PhonebookProfile.PHONEBOOK_PROFILE_TYPE_UID);
     }
 
     @Override
     public Collection<ProfileType> getProfileTypes(@Nullable Locale locale) {
-        return Collections.singleton(PhonebookProfile.PHONEBOOK_PROFILE_TYPE);
+        return Set.of(createLocalizedProfileType(PhonebookProfile.PHONEBOOK_PROFILE_TYPE, locale));
+    }
+
+    private ProfileType createLocalizedProfileType(ProfileType profileType, @Nullable Locale locale) {
+        final LocalizedKey localizedKey = getLocalizedProfileTypeKey(profileType.getUID(), locale);
+
+        final ProfileType cachedlocalizedProfileType = localizedProfileTypeCache.get(localizedKey);
+        if (cachedlocalizedProfileType != null) {
+            return cachedlocalizedProfileType;
+        }
+
+        final ProfileType localizedProfileType = localize(profileType, locale);
+        if (localizedProfileType != null) {
+            localizedProfileTypeCache.put(localizedKey, localizedProfileType);
+            return localizedProfileType;
+        } else {
+            return profileType;
+        }
+    }
+
+    private @Nullable ProfileType localize(ProfileType profileType, @Nullable Locale locale) {
```

Comment:
Why do you need these two methods? If I didn't overlook something, `localize` and `getLocalizedProfileTypeKey` are only called once.

Reply:
Absolutely right. It is a developer's sin. Copy & paste from OHC `SystemProfileFactory`.